docs/guidance/pod-security.md

```diff
@@ -0,0 +1,96 @@
+kind create cluster --config kind-config.yaml --image=kindest/node:v1.24.0
+```
+The `kind-config.yaml` enables audit logging with the audit policy defined in `audit-policy.yaml`. The `audit-policy.yaml`
+defines an auditing policy to listen to the Pod events in the namespace `pod-security`. With this policy, we can check
+whether our Pods violate the policies in `restricted` standard or not.
+
+The feature [Pod Security Admission](https://kubernetes.io/docs/concepts/security/pod-security-admission/) is firstly 
+introduced in Kubernetes v1.22 (alpha) and becomes stable in Kubernetes v1.25. In addition, KubeRay currently supports 
+Kubernetes from v1.19 to v1.24. (At the time of writing, we have not tested KubeRay with Kubernetes v1.25). Hence, I use **Kubernetes v1.24** in this step.
+
+# Step 2: Check the audit logs
+```bash
+docker exec kind-control-plane cat /var/log/kubernetes/kube-apiserver-audit.log
+```
+The log should be empty because the namespace `pod-security` does not exist.
+
+# Step 3: Create the `pod-security` namespace
+```bash
+kubectl create ns pod-security
+kubectl label --overwrite ns pod-security \
+  pod-security.kubernetes.io/warn=restricted \
+  pod-security.kubernetes.io/warn-version=latest \
+  pod-security.kubernetes.io/audit=restricted \
+  pod-security.kubernetes.io/audit-version=latest \
+  pod-security.kubernetes.io/enforce=restricted \
+  pod-security.kubernetes.io/enforce-version=latest
+```
+With the `pod-security.kubernetes.io` labels, the built-in Kubernetes Pod security admission controller will apply the 
+`restricted` Pod security standard to all Pods in the namespace `pod-security`. The label
+`pod-security.kubernetes.io/enforce=restricted` means that the Pod will be rejected if it violate the policies defined in 
+`restricted` security standard. See [Pod Security Admission](https://kubernetes.io/docs/concepts/security/pod-security-admission/) for more details about the labels.
+
+# Step 4: Install the KubeRay operator
+```bash
+# Path: helm-chart/kuberay-operator
+helm install -n pod-security kuberay-operator .
+```
+
+# Step 5: Create a RayCluster (Choose either Step 5.1 or Step 5.2)
+* If you choose Step 5.1, no Pod will be created in the namespace `pod-security`.
+* If you choose Step 5.2, Pods can be created successfully.
+
+## Step 5.1: Create a RayCluster without proper `securityContext` configurations
+```bash
+# Path: ray-operator/config/samples
+kubectl apply -n pod-security -f ray-cluster.complete.yaml
+
+# Wait 20 seconds and check audit logs for the error messages.
+docker exec kind-control-plane cat /var/log/kubernetes/kube-apiserver-audit.log
+
+# Example error messagess
+# "pods \"raycluster-complete-head-fkbf5\" is forbidden: violates PodSecurity \"restricted:latest\": allowPrivilegeEscalation != false (container \"ray-head\" must set securityContext.allowPrivilegeEscalation=false) ...
+```
+No Pod will be created in the namespace `pod-security`, and check audit logs for error messages.
+
```

Comment:
Let's add the instruction to delete the faulty raycluster.

Reply:
Good idea. I will update it later.

Reply:
Also, we can add `kubectl get pod -n pod-security` with output showing only the operator.

Reply:
Updated. https://github.com/ray-project/kuberay/pull/750/commits/5e0ff9d98d197ff50deb82c9139ea819b78bb644